retail/2.7/modules/ROOT/pages/sfdc-customization-guide.adoc

```diff
@@ -2,30 +2,429 @@
+| Y
+| Y
+
+| Account
+| Customer Status
+| Customer_Status
+| Picklist
+| -
+| N
+| N
+
+| Managed_External_Id__c
+| External Id
+| External_Id
+| Text
+| 80
+| N
+| N
+
+| Managed_External_Id__c
+| External Id Type
+| External_Id_Type
+| Picklist
+| -
+| N
+| N
+
+| Managed_External_Id__c
+| Object Id
+| Object_Id
+| Text
+| 80
+| N
+| N
+
+| Managed_External_Id__c
+| Object Type
+| Object_Type
+| Picklist
+| -
+| N
+| N
+
+| Managed_External_Id__c
+| Status
+| Status
+| Picklist
+| -
+| N
+| N
+
+| Managed_External_Id__c
+| Status Last Changed
+| Status_Last_Changed
+| Date/Time
+| -
+| N
+| N
+
+| Contact
+| Global_Individual_Id
+| Global_Individual_Id
+| Text
+| 36
+| Y
+| Y
+
+| Order
+| Global_SalesOrder_Id
+| Global_SalesOrder_Id
+| Text
+| 36
+| Y
+| Y
+
+| Product2
+| Global_Product_Id
+| Global_Product_Id
+| Text
+| 36
+| Y
+| Y
+
 |===
 
+*Unique fields should be case-sensitive
+
+=== Create via Apex Script
+
+An Apex class called `CustomFieldUtility` can be used to create custom fields programmatically. The source for this class can be found in the https://anypoint.mulesoft.com/exchange/org.mule.examples/accelerator-common-resources-src[Accelerator Common Resources^] project. Download this project and follow the steps below.
+
+. Click on Salesforce Setup icon and select `Developer Console` to open a new console window
+. Go to `+File -> New -> Apex class+` and create a new class named `CustomFieldUtility`
+. Copy the code from `accel-common-resources/salesforce/CustomFieldUtility.apxc` to the script editor
+. Select the `+File -> Save+` menu option to compile and save the class
+. Select `+Debug -> Open Execute Anonymous Window+`
+. Copy and paste the contents of `accel-common-resources/salesforce/CreateCustomFields.txt` into the window.
+. Enable the `Open Log` option and click the *Execute* button.
+
+Once the script completes, select the `Debug Only` filter option to show just the results of each create request. Verify that all fields have been successfully created.
+
+=== Create Manually via Salesforce
+
+To create each of these custom fields and enable visibility for them on page layouts:
+
+. Click the Salesforce Setup icon.
+. Select the `Object Manager` tab from the Setup home page.
+. Find and select the target Salesforce object.
+. Select the `Fields & Relationships` page.
+. Click the *New* button and create the field as specified above.
+. Once the field has been created, click the `Set Field-Level Security` button (or do these steps later - see below).
+. Enable for desired profile, or tick the checkbox next to `Visible` to enable visibility for the desired profile(s).
+
+Repeat these steps for each custom field in the above list.
+
+TIP: Instead of adjusting permissions for each field as you go, if you are only making them available to one or two profiles it may be more efficient to go to the `+Users -> Profiles -> {profile} -> Object Settings+` for each object and adjust the permissions for multiple fields at once.
```

Comment:
```suggestion
[NOTE]
Instead of adjusting permissions for each field as you go, if you are only making them available to one or two profiles, then it may be more efficient to go to `Users > Profiles > {profile} > Object Settings` for each object and adjust the permissions for multiple fields at once.
```